traitsui/tests/test_ui.py

```diff
@@ -190,25 +193,26 @@ def test_destroy_after_ok_qt(self):
         from pyface import qt
 
         foo = FooDialog()
-        ui = foo.edit_traits()
+        with store_exceptions_on_all_threads(), create_ui(foo) as ui:
 
-        # keep reference to the control to check that it was deleted
-        control = ui.control
+            # keep reference to the control to check that it was deleted
+            control = ui.control
 
-        # decorate control's `deleteLater` function to check that it is called
-        control.deleteLater = count_calls(control.deleteLater)
+            # decorate control's `deleteLater` function to check that it is
+            # called
+            control.deleteLater = count_calls(control.deleteLater)
 
-        # press the OK button and close the dialog
-        okb = control.findChild(qt.QtGui.QPushButton)
-        okb.click()
+            # press the OK button and close the dialog
+            okb = control.findChild(qt.QtGui.QPushButton)
+            okb.click()
 
-        self.assertIsNone(ui.control)
-        self.assertEqual(control.deleteLater._n_calls, 1)
+            self.assertIsNone(ui.control)
+            self.assertEqual(control.deleteLater._n_calls, 1)
 
     @skip_if_null
     def test_no_spring_trait(self):
         obj = DisallowNewTraits()
-        ui = obj.edit_traits()
-        ui.dispose()
+        with store_exceptions_on_all_threads(), create_ui(obj):
```

Comment:
The additional `store_exceptions_on_all_threads` here (and a couple of other tests) are drive-by and orthogonal. Now I am regretting mixing them in.  Will remove them again to keep this PR focus.